### PR TITLE
Fixes #10 - Smooth out the lines

### LINKED ISSUE #10: Smooth out the lines
Can use a library like this:
http://paperjs.org/tutorials/paths/working-with-path-items/

### PR DESCRIPTION
**Pull request by andrewheins**
I've created a basic implementation of line smoothing. Hopefully this helps.

### CODE REVIEW

File: opentok-whiteboard.js [hunk 1]
```diff
@@ -236,21 +217,47 @@ var OpenTokWhiteboard = angular.module('opentok-whiteboard', ['opentok'])
...
+                    // Where the line
                     client.lastX = x;
                     client.lastY = y;
+                    
+                    if( window.path ) {
+                        window.path.selected = false;
+                    }
+                    
+                    // Create a new path object
+                    window.path = new paper.Path();
```
Comment (aullman) [inline opentok-whiteboard.js]: The `window.path()` doesn't get created until the current user starts drawing. This means that if you receive the events from someone else drawing then you get an error about window.path being undefined.

File: opentok-whiteboard.js [hunk 2]
```diff
@@ -55,19 +59,20 @@ var OpenTokWhiteboard = angular.module('opentok-whiteboard', ['opentok'])
...
                             {'background-color': 'brown'}];
             scope.captureText = iOS ? 'Email' : 'Capture';
+            
+            scope.strokeCap = 'round';
+            scope.strokeJoin = 'round';
 
             canvas.width = attrs.width || element.width();
             canvas.height = attrs.height || element.height();
             
+            if (!ctx) {
```
Comment (aullman) [inline opentok-whiteboard.js]: This `ctx` variable isn't used anymore is it?

**Comment by aullman** [comment]
Thanks for this @andrewheins. I would love to merge it but it seems that there are some bugs with collaborative drawing.

**Comment by bostondv** [comment]
@aullman Andrew and I worked on this together. After submitting this PR we realized there were some major problems with the collaborative drawing. It worked great just drawing by yourself :(

We're kind of stuck and not sure how to fix the problems. Any chance you'd have some time to help us out with making it work collaboratively?

**Comment by sumansaurabh** [comment]
May I know the issue you faced.

**Comment by bostondv** [comment]
Hey @sumansaurabh, that would be awesome! 

It works fine mostly with only one person in the meeting but as soon as another starts using the whiteboard thats where the problems come in.

The various problems in @andrewheins master branch that I know about are:

- Not all lines get synced between users (or are slow to appear)
- Lines drawn from other users appear connected to each other and your lines when synced
- Color is not properly transmitted between lines
- Erasing between users doesn't work (sometimes actually appears as lines to other user instead of erasing)
- Undo/redo doesn't sync between users

Those are some I know about, though there might be other problems we haven't realized. We also have a "fullDraw" branch that we experimented with trying to fix colors/undo/redo but it's currently not working either.

**Pull request closed without merging**